src/test/client/TestNewOpsClient.h

```diff
@@ -0,0 +1,105 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2022 Red Hat
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "gtest/gtest.h"
+
+#include "common/async/context_pool.h"
+#include "global/global_context.h"
+
+#include "msg/Messenger.h"
+#include "mon/MonClient.h"
+#include "osdc/ObjectCacher.h"
+#include "client/MetaRequest.h"
+#include "client/Client.h"
+
+#define dout_subsys ceph_subsys_client
+
+class TestNewOpsClient : public Client {
```

Comment:
Move definition to `TestClient.h` and modify `class TestClient` to instantiate this class. So it's common to all tests in `src/test/client/`.

Reply:
This will also obsolete `TestNewOps` below.

Reply:
I was thinking of having a template for client instance in `TestClient.h` but It throws error `object of abstract class type "TestClient<TestNewOpsClient>" is not allowed:C/C++(322) ops.cc(21, 1): pure virtual function "testing::Test::TestBody" has no overrider`. Seems like it's not allowing template because the base class in `gtest.h` has pure virtual function. Do you know anyway we can achieve something like this?

Reply:
Why do you need to have a template for the `Client` testing subclass? I think you can just have a single concrete subclass common for all tests in `src/test/client/`.

Reply:
yeah makes sense, class `TestNewOpsClient` already inherits `Client`, instantiating it instead of `Client` would work

Reply:
its done. thanks